siliconcompiler/project.py

```
@@ -1339,23 +1339,31 @@ def show(self, filename: Optional[str] = None, screenshot: bool = False,
                     self.logger.error(f"Extension '{extension}' not supported by "
-                                      f"registered showtools: {', '.join(exts)}")
+                                      f"registered showtools: {', '.join(sorted(set(all_exts)))}")
                     return None
-                exts = [extension]
+                # Search for the specific extension only, respecting tool order
+                search_exts = [(cls, ext) for cls, ext in search_exts if ext == extension]
 
-            for ext in exts:
+            # Search for files in tool-ordered sequence
+            # This ensures that earlier-registered (higher-priority) tools are tried first
+            for task_cls, ext in search_exts:
```

Ruff (0.15.11)

[warning] Loop control variable `task_cls` not used within loop body

Rename unused `task_cls` to `_task_cls`

(B007)